src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -238,5 +239,22 @@ public WrappedApplicationEnvironment(string applicationBasePath, IApplicationEnv
 
             public FrameworkName RuntimeFramework { get; }
         }
+
+        private static string ResolvePath(string applicationBasePath)
+        {
+            if (Path.IsPathRooted(applicationBasePath))
+            {
+                return applicationBasePath;
+            }
+
+            var basePath =
+#if NET451
+                (string)AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY") ??
+                AppDomain.CurrentDomain.BaseDirectory;
+#else
+                AppContext.BaseDirectory;
+#endif
+            return Path.Combine(Path.GetFullPath(basePath), applicationBasePath);
```

Comment:
Fair point. As per our conversation I will do GetFullPath on the combined result to collapse unneeded traversals.